core/network/impl/peer_manager_impl.cpp

```diff
@@ -746,38 +744,20 @@ namespace kagome::network {
       return;
     }
     if (connection->isInitiator()) {
-      auto out_peers_count = std::count_if(
-          active_peers_.begin(), active_peers_.end(), [](const auto &el) {
-            return el.second.peer_type == PeerType::PEER_TYPE_OUT;
-          });
-      if (out_peers_count > app_config_.outPeers()) {
+      if (countPeers(true) >= app_config_.outPeers()) {
         connecting_peers_.erase(peer_id);
         disconnectFromPeer(peer_id);
         return;
       }
     } else {
-      auto in_peers_count = 0u;
-      auto in_light_peers_count = 0u;
       if (peer_states_[peer_id].roles.flags.full == 1) {
-        for (const auto &peer : active_peers_) {
-          if (peer.second.peer_type == PeerType::PEER_TYPE_IN
-              and peer_states_[peer.first].roles.flags.full == 1) {
-            ++in_peers_count;
-          }
-        }
-        if (in_peers_count >= app_config_.inPeers()) {
+        if (countPeers(false) >= app_config_.inPeers()) {
           connecting_peers_.erase(peer_id);
           disconnectFromPeer(peer_id);
           return;
         }
       } else if (peer_states_[peer_id].roles.flags.light == 1) {
-        for (const auto &peer : active_peers_) {
-          if (peer.second.peer_type == PeerType::PEER_TYPE_IN
-              and peer_states_[peer.first].roles.flags.light == 1) {
-            ++in_light_peers_count;
-          }
-        }
-        if (in_light_peers_count >= app_config_.inPeersLight()) {
+        if (countPeers(false, true) >= app_config_.inPeersLight()) {
```

Comment:
Can we use `enum` for argument? It would make call more readable

Reply:
No